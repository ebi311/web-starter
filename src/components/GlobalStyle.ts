import { createGlobalStyle } from 'styled-components';
import theme from './Theme';

export const GlobalStyle = createGlobalStyle`
  html, body {
    background-color: ${theme.background};
    color: ${theme.foreground};
    font-size: ${theme.fontSize}px;
    font-family: 'BIZ UDPGothic', 'Meiryo UI', 'Yu Gothic', Verdana, Arial, Helvetica, Roboto, sans-serif;
  }
  html, body, div, p {
    margin: 0;
    padding: 0;
  }
  :focus {
    outline: none;
  }
    /*スクロールバー全体*/
  ::-webkit-scrollbar {
    width: 4px;
  }
  /*スクロールバーの軌道*/
  ::-webkit-scrollbar-track-piece {
  }

  /*スクロールバーの動く部分*/
  ::-webkit-scrollbar-thumb {
    background-color: ${theme.surface1};
    border-radius: 2px;
  }

`;
