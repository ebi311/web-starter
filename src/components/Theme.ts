import { darken, lighten, rgba } from 'polished';
import baseStyled, { ThemedStyledInterface } from 'styled-components';

const _theme = {
  background: '#fff',
  surface: '#757575',
  foreground: '#424242',
  primary0: '#311b92',
  primary1: '#6746c3',
  primary2: '#000063',
  secondary0: '#fbc02d',
  secondary1: '#fff263',
  secondary2: '#c49000',
  fontSize: 16,
  griSize: 8,
  halfGridSize: 4,
};

const theme = {
  ..._theme,
  surface1: lighten(0.2, _theme.surface),
  surface2: darken(0.2, _theme.surface),
  foreground1: rgba(_theme.foreground, 0.5),
};

export default theme;

export type Theme = typeof theme;

export const styled = baseStyled as ThemedStyledInterface<Theme>;
