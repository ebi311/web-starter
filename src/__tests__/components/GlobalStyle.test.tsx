import React from 'react';
import { GlobalStyle } from '../../components/GlobalStyle';
import { shallow } from 'enzyme';

describe('GlobalStyle', () => {
  test('success', () => {
    const wrapper = shallow(<GlobalStyle />);
    expect(wrapper).toMatchSnapshot();
  });
});
