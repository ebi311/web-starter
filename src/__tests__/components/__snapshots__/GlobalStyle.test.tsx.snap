// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStyle success 1`] = `""`;
